src/MiniJuvix/Translation/AbstractToMicroJuvix.hs

```diff
@@ -183,11 +183,23 @@ goIden i = case i of
   A.IdenAxiom a -> IdenAxiom (goName (a ^. A.axiomRefName))
   A.IdenInductive a -> IdenInductive (goName (a ^. A.inductiveRefName))
 
+goExpressionFunction :: forall r. A.Function -> Sem r FunctionExpression
+goExpressionFunction f = do
+  l' <- goParam (f ^. A.funParameter)
+  r' <- goExpression (f ^. A.funReturn)
+  return (FunctionExpression l' r')
+  where
+    goParam :: A.FunctionParameter -> Sem r Expression
```

Comment:
I don't fully understand this one. Is this case needed because in a type like `(x : A) → B,` the notation `(x : A)` is not supported?

Reply:
that's right